extensions/smallrye-metrics/deployment/src/main/java/io/quarkus/smallrye/metrics/deployment/SmallRyeMetricsProcessor.java

```diff
@@ -495,6 +507,12 @@ void registerMetricsFromProducers(
                 }
             }
         }
+
+        for (MetricsFactoryConsumerBuildItem item : metricsFactoryConsumerBuildItems) {
```

Comment:
Why is this inside the method `registerMetricsFromProducers`? Metrics from producers will go away in MP Metrics 3.0 and so will this method.

Reply:
Ah. I will rename that method -- the reason is mostly parameters (that takes place at runtime init after bean validation, which means the beans that some of the consumers will need will all be resolved (ValidationPhaseBuildItem / BeanArchiveIndexBuildItem)